tests/integration/test_data/npm_lockfile3_yarn_registry/bom.json

```diff
@@ -0,0 +1,409 @@
+      "type": "library",
+      "version": "0.2.0"
+    },
+    {
+      "name": "fresh",
+      "purl": "pkg:npm/fresh@0.5.2",
+      "type": "library",
+      "version": "0.5.2"
+    },
+    {
+      "name": "function-bind",
+      "purl": "pkg:npm/function-bind@1.1.1",
+      "type": "library",
+      "version": "1.1.1"
+    },
+    {
+      "name": "get-intrinsic",
+      "purl": "pkg:npm/get-intrinsic@1.2.1",
+      "type": "library",
+      "version": "1.2.1"
+    },
+    {
+      "name": "has-proto",
+      "purl": "pkg:npm/has-proto@1.0.1",
+      "type": "library",
+      "version": "1.0.1"
+    },
+    {
+      "name": "has-symbols",
+      "purl": "pkg:npm/has-symbols@1.0.3",
+      "type": "library",
+      "version": "1.0.3"
+    },
+    {
+      "name": "has",
+      "purl": "pkg:npm/has@1.0.3",
+      "type": "library",
+      "version": "1.0.3"
+    },
+    {
+      "name": "http-errors",
+      "purl": "pkg:npm/http-errors@2.0.0",
+      "type": "library",
+      "version": "2.0.0"
+    },
+    {
+      "name": "iconv-lite",
+      "purl": "pkg:npm/iconv-lite@0.4.24",
+      "type": "library",
+      "version": "0.4.24"
+    },
+    {
+      "name": "inherits",
+      "purl": "pkg:npm/inherits@2.0.4",
+      "type": "library",
+      "version": "2.0.4"
+    },
+    {
+      "name": "ipaddr.js",
+      "purl": "pkg:npm/ipaddr.js@1.9.1",
+      "type": "library",
+      "version": "1.9.1"
+    },
+    {
+      "name": "media-typer",
+      "purl": "pkg:npm/media-typer@0.3.0",
+      "type": "library",
+      "version": "0.3.0"
+    },
+    {
+      "name": "merge-descriptors",
+      "purl": "pkg:npm/merge-descriptors@1.0.1",
+      "type": "library",
+      "version": "1.0.1"
+    },
+    {
+      "name": "methods",
+      "purl": "pkg:npm/methods@1.1.2",
+      "type": "library",
+      "version": "1.1.2"
+    },
+    {
+      "name": "mime-db",
+      "purl": "pkg:npm/mime-db@1.52.0",
+      "type": "library",
+      "version": "1.52.0"
+    },
+    {
+      "name": "mime-types",
+      "purl": "pkg:npm/mime-types@2.1.35",
+      "type": "library",
+      "version": "2.1.35"
+    },
+    {
+      "name": "mime",
+      "purl": "pkg:npm/mime@1.6.0",
+      "type": "library",
+      "version": "1.6.0"
+    },
+    {
+      "name": "ms",
+      "purl": "pkg:npm/ms@2.0.0",
+      "type": "library",
+      "version": "2.0.0"
+    },
+    {
+      "name": "ms",
+      "purl": "pkg:npm/ms@2.1.2",
+      "type": "library",
+      "version": "2.1.2"
+    },
+    {
+      "name": "ms",
+      "purl": "pkg:npm/ms@2.1.3",
+      "type": "library",
+      "version": "2.1.3"
+    },
+    {
+      "name": "negotiator",
+      "purl": "pkg:npm/negotiator@0.6.3",
+      "type": "library",
+      "version": "0.6.3"
+    },
+    {
+      "name": "npm_test",
+      "purl": "pkg:npm/npm_test@1.1.0?vcs_url=git%2Bhttps://github.com/cachito-testing/npm-cachi2-registry-yarnpkg.git%40f2daad2b7a844779bdae8b3b66e6ec1c9de70beb",
```

Comment:
nitpick: the name of the main package is different between lockfile v1 and lockfile v3